explainer.md

```diff
@@ -215,11 +224,189 @@ defaults.
 ```js
 const config_allow = {
   allowElements: [ "div", "p", "em", "b" ]  // Allows only those four elements.
-      // Output with "safe" and "unsafe" methods should be the same.
+  allowAttributes: [ "class" ]  // Allows only class attributes.
+      // Output with "safe" and "unsafe" methods are the same for this config.
 };
 const config_block = {
-  blockElements: [ "style" ]  // Allows a lot of things. But not <style>.
+  blockElements: [ "style" ],  // Allows a lot of things. But not <style>.
+  blockAttributes: [ "class", "style" ]  // No style or class attribute either.
       // And not XSS-y stuff, either, if used with a "safe" method.
       // Output with "safe" and "unsafe" methods might be quite different.
 };
 ```
+
+### Configuration List-Types: Allow-lists, Blocks-lists, and Drop-lists
+
+For elements there are three types of lists: Allow-lists, Block-lists
+and Drop-lists. These correspond to three related but somehow different use
+cases:
+
+If you have a specific set of functionality in mind that you wish to
+retain and would like to drop everything else as a matter of caution, this
+can be accomplished with allow lists. This is probably the most common case.
+For example, you might want allow user comments with simple styling, but
+no style definitions:
+
+```js
+const config_only_simple_styling = {
+  allowElements: ["p", "span", "em", "b", "i", "u", "s", "img"],
+  allowAttributes: ["class", "src"],
+};
+```
+
+If you wish to let most content pass through, but have a specific set of
+functionality in mind that you wish to also block in your application, you can
+use block-lists. For example, you might want to chiefly drop XSS-y content, but
+also disallow new styles or `id`s not already declared in your document:
+
+```js
+const config_anything_but_style_definitions = {
+  blockElements: ["style"],
+  blockAttributes: ["style", "id"],
+};
+```
+
+If you know your input will contain elaborate markup and you only wish to
+retain some of it, but you want to retain the full textual content, you
+can use `dropElement`. This will remove only elements but retain their children.
+There is no `dropAttribute` because attribute nodes do not have children.
+
+```js
+const config_that_drops_things = {
+  dropElements: ["span", "em", "u", "s", "i", "b"]
+};
+```
+
+How do multiple element list types interact? In order for an element to
+pass, it must pass _all_ lists.
+
+You can use allow-lists and block-lists together, but note that this doesn't
+usually make much sense since any combination of allow- and block-lists can
+be rewritten as an allow-lists of equal or shorter length by removing any
+block-list element from the allow-list, or dropping it altogether if there
+was no matching allow-list entry in the first place.
+
+```js
+// equivalent to:  { allowElements: ["i"] }
+const config_that_mixes_allow_and_block_lists = {
+    allowElements: ["i", "u"],
+    blockElements: ["u", "s"],
+};
```

Comment:
That's all correct and matches my understanding. I think it would make more sense for a reader if we defined and then pointed to the processing model (Roughly "check first drop / block first, then allow." like we already have in the "determine the sanitize action for an element" algorithm)